File: commons/javafx/src/main/java/com/cedarsoft/commons/javafx/NumberStringConverterForIntegers.java
package com.cedarsoft.commons.javafx;

import java.text.NumberFormat;
import java.util.Locale;

import javax.annotation.Nonnull;

import javafx.util.converter.NumberStringConverter;

/**
 * A {@link NumberStringConverter} for integer and long values.
 *
GlobalTilesCache */
public class NumberStringConverterForIntegers extends NumberStringConverter {
  @Nonnull
  private final NumberFormat integerNumberFormat; // works for Integers _and_ Longs

  public NumberStringConverterForIntegers() {
    this(Locale.getDefault());
  }

  public NumberStringConverterForIntegers(@Nonnull Locale locale) {
    integerNumberFormat = NumberFormat.getIntegerInstance(locale); // works for Integers _and_ Longs
  }

  @Nonnull
  @Override
  protected NumberFormat getNumberFormat() {
    return integerNumberFormat;
  }

  @Override
  public Number fromString(String value) {
    if ("-".equals(value)) {
      return 0;
    }
    return super.fromString(value);
  }
}

File: commons/javafx/src/main/kotlin/it/neckar/open/javafx/NumberStringConverterForIntegers.kt
package it.neckar.open.javafx

import javafx.util.converter.NumberStringConverter
import java.text.NumberFormat
import java.util.Locale
import javax.annotation.Nonnull

/**
 * A [NumberStringConverter] for integer and long values.
 *
 * GlobalTilesCache  */
class NumberStringConverterForIntegers
@JvmOverloads constructor(@Nonnull locale: Locale? = Locale.getDefault()) : NumberStringConverter() {

  // works for Integers _and_ Longs
  private val integerNumberFormat: NumberFormat = NumberFormat.getIntegerInstance(locale)

  override fun getNumberFormat(): NumberFormat {
    return integerNumberFormat
  }

  override fun fromString(value: String?): Number? {
    return if ("-" == value) {
      0
    } else super.fromString(value)
  }
}
